aten/src/ATen/native/cuda/ForeachTensorAddList.cu

```diff
@@ -0,0 +1,47 @@
+#include <ATen/Dispatch.h>
+#include <ATen/native/ForeachUtils.h>
+#include <ATen/native/cuda/ForeachFunctors.cuh>
+
+namespace at { namespace native {
+
+std::vector<Tensor> foreach_tensor_add_list_kernel_cuda(TensorList tensors1, TensorList tensors2) {
+    verify_list(tensors1, tensors2);
+
+    if (!check_fast_route(tensors1, tensors2)) {
+        return at::native::foreach_tensor_add_list_kernel_slow(tensors1, tensors2);
+    }
+
+    std::vector<std::vector<at::Tensor>> tensor_lists; 
+    std::vector<at::Tensor> vec_res;
```

Comment:
I've been told it is good practice to reserve space for vectors beforehand to avoid the vector re-sizing itself as elements are added. See https://www.geeksforgeeks.org/using-stdvectorreserve-whenever-possible/.

This would look like:
```
std::vector<at::Tensor> vec_res;
vec_res.reserve(tensors1.size());
```